sources/packages/backend/workflow/test/2023-2024/parttime-assessment/parttime-assessment-2023-2024-eligibility-BCAG.e2e-spec.ts

```diff
@@ -0,0 +1,75 @@
+import { PROGRAM_YEAR } from "../constants/program-year.constants";
+import {
+  createFakeConsolidatedPartTimeData,
+  executePartTimeAssessmentForProgramYear,
+} from "../../test-utils";
+import { YesNoOptions } from "@sims/test-utils";
+import { CredentialType, InstitutionTypes } from "../../models";
+
+describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-eligibility-BCAG.`, () => {
+  it(
+    "Should determine BCAG as eligible when total assessed need is greater than 0 " +
+      `, institution type is ${InstitutionTypes.BCPublic}, total family income is less than the threshold ` +
+      `and program credential type is ${CredentialType.UnderGraduateDegree} for separated student.`,
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataRelationshipStatus = "other";
+      assessmentConsolidatedData.studentDataDependantstatus = "independant";
+      assessmentConsolidatedData.institutionType = InstitutionTypes.BCPublic;
+      assessmentConsolidatedData.programCredentialType =
+        CredentialType.UnderGraduateDegree;
+
+      // Act
+      const calculatedAssessment =
+        await executePartTimeAssessmentForProgramYear(
+          PROGRAM_YEAR,
+          assessmentConsolidatedData,
+        );
+
+      // Assert
+      expect(calculatedAssessment.variables.awardEligibilityBCAG).toBe(true);
+      expect(
+        calculatedAssessment.variables.finalProvincialAwardNetBCAGAmount,
+      ).toBeGreaterThan(0);
+    },
+  );
+
+  it(`Should determine BCAG as not eligible when institution type is ${InstitutionTypes.BCPrivate} for a married student.`, async () => {
```

Comment:
Does "married student" impact the output of the E2E scenario? If not I do not see a reason to make it part of the description or the test. If the point is about the family size, it may be explicitly expressed.

Reply:
https://github.com/bcgov/SIMS/pull/2506#discussion_r1394937155